content/en/tracing/trace_collection/library_config/nodejs.md

```diff
@@ -26,87 +26,211 @@ Tracer settings can be configured with the following environment variables:
 **Example:**<br>
   - Set the span sample rate to 50% for the service `my-service` and operation name `http.request`, up to 50 traces per second: `'[{"service": "my-service", "name": "http.request", "sample_rate":0.5, "max_per_second": 50}]'`
 
+`DD_SPAN_SAMPLING_RULES_FILE`
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+Points to a JSON file that contains the span sampling rules. `DD_SPAN_SAMPLING_RULES` takes precedence over this variable.
+
 `DD_RUNTIME_METRICS_ENABLED`
-: **Default**:  `false`<br>
+: **Configuration**: `runtimeMetrics`<br>
+**Default**: `false`<br>
 Whether to enable capturing runtime metrics. Port `8125` (or configured with `DD_DOGSTATSD_PORT`) must be opened on the Agent for UDP.
 
 `DD_SERVICE_MAPPING`
-: **Example**: `mysql:my-mysql-service-name-db,pg:my-pg-service-name-db`<br>
+: **Configuration**: `serviceMapping`<br>
+**Default**: N/A<br>
+**Example**: `mysql:my-mysql-service-name-db,pg:my-pg-service-name-db`<br>
 Provide service names for each plugin. Accepts comma separated `plugin:service-name` pairs, with or without spaces.
 
 `DD_TRACE_DISABLED_PLUGINS`
-: A comma-separated string of integration names automatically disabled when the tracer is initialized. Environment variable only, for example, `DD_TRACE_DISABLED_PLUGINS=express,dns`.
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+**Example**: `DD_TRACE_DISABLED_PLUGINS=express,dns`<br>
+A comma-separated string of integration names automatically disabled when the tracer is initialized.
 
 `DD_TRACE_LOG_LEVEL`
-: **Default**: `debug`<br>
+: **Configuration**: `logLevel`<br>
+**Default**: `debug`<br>
 A string for the minimum log level for the tracer to use when debug logging is enabled, for example, `error`, `debug`.
 
-
-For more options including the programmatic configuration API, see the [API documentation][2].
+Flush Interval
+: **Configuration**: `flushInterval`<br>
+**Default**: `2000`<br>
+Interval in milliseconds at which the tracer will submit traces to the agent.
+
+`DD_TRACE_PARTIAL_FLUSH_MIN_SPANS`
+: **Configuration**: `flushMinSpans`<br>
+**Default**: `1000`<br>
+Number of spans before partially exporting a trace. This prevents keeping all the spans in memory for very large traces.
+
+`DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP`
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+A regex to redact sensitive data from incoming requests' query string reported in the `http.url` tag (matches will be replaced with `<redacted>`). Can be an empty string to disable redaction or `.*` to redact all query string. **WARNING: this regex will execute for every incoming request on an unsafe input (url), please make sure you use a safe regex.** |
+
+`DD_TRACE_CLIENT_IP_HEADER`
+: **Configuration**: N/A<br>
+**Default**: N/A<br>
+Custom header name to source the `http.client_ip` tag from.
+
+DNS Lookup Function
+: **Configuration**: `lookup`<br>
+**Default**: `require('dns').lookup`<br>
+Custom function for DNS lookups when sending requests to the agent.
```

Comment:
What is the reason for exposing this? Do we actually _want_ this documented?

Reply:
Good question for @rochdev e8e503a90e65bbc6f5f407c26e9a02b51576ba63.

Reply:
[The PR](https://github.com/DataDog/dd-trace-js/pull/1003) clarifies it.

Reply:
Maybe we can take some of the explanation from Roch to make the purpose of this setting clearer? If _I_ don't know what it's for and I literally work on this library, I can't imagine some random dev reading the docs and having any idea why they might want to use that. 😅

Reply:
I've updated this with content based on Roch's original explanation.